specs/phase0/fork-choice.md

```diff
@@ -272,25 +334,87 @@ def get_head(store: Store) -> Root:
         head = max(children, key=lambda root: (get_weight(store, root), root))
 ```
 
-#### `should_update_justified_checkpoint`
+#### `update_checkpoints`
 
 ```python
-def should_update_justified_checkpoint(store: Store, new_justified_checkpoint: Checkpoint) -> bool:
+def update_checkpoints(store: Store, justified_checkpoint: Checkpoint, finalized_checkpoint: Checkpoint) -> None:
     """
-    To address the bouncing attack, only update conflicting justified
-    checkpoints in the fork choice if in the early slots of the epoch.
-    Otherwise, delay incorporation of new justified checkpoint until next epoch boundary.
+    Update checkpoints in store if necessary
+    """
+    # Update justified checkpoint
+    if justified_checkpoint.epoch > store.justified_checkpoint.epoch:
```

Comment:
Sanity check: updating these two independent is safe because in `on_block` and any other entry mechanism, we are pre-filtering by finalized anyway so justified will remain coherent with finality?

Said another way, we would reject blocks from non-finality-coherent chains before we got to this point and thus not get to a place where a higher justified from a non-finalized ancestor would work its way into the store

Reply:
Yes, that's correct.